front/lib/resources/user_resource.ts

```diff
@@ -0,0 +1,250 @@
+  }
+
+  static async fetchByEmail(email: string): Promise<UserResource | null> {
+    const user = await User.findOne({
+      where: {
+        email,
+      },
+    });
+
+    return user ? new UserResource(User, user.get()) : null;
+  }
+
+  static async fetchByProvider(
+    provider: UserProviderType,
+    providerId: string
+  ): Promise<UserResource | null> {
+    const user = await User.findOne({
+      where: {
+        provider,
+        providerId,
+      },
+    });
+
+    return user ? new UserResource(User, user.get()) : null;
+  }
+
+  static async getWorkspaceFirstAdmin(
+    workspaceId: number
+  ): Promise<UserResource | null> {
+    const user = await User.findOne({
+      include: [
+        {
+          model: MembershipModel,
+          where: {
+            role: "admin",
+            workspaceId,
+          },
+          required: true,
+        },
+      ],
+      order: [["createdAt", "ASC"]],
+    });
+
+    return user ? new UserResource(User, user.get()) : null;
+  }
+
+  async updateAuth0Sub(sub: string): Promise<void> {
+    await this.model.update(
+      {
+        auth0Sub: sub,
+      },
+      {
+        where: {
+          id: this.id,
+        },
+      }
+    );
+  }
+
+  async updateName(
+    firstName: string,
+    lastName: string | null
+  ): Promise<Result<undefined, Error>> {
+    try {
+      await this.model.update(
+        {
+          firstName,
+          lastName,
+        },
+        {
+          where: {
+            id: this.id,
+          },
+          returning: true,
+        }
+      );
+
+      return new Ok(undefined);
+    } catch (err) {
+      return new Err(err as Error);
+    }
+  }
+
+  async updateInfo(
+    username: string,
+    firstName: string,
+    lastName: string | null,
+    email: string
+  ): Promise<void> {
+    const [, affectedRows] = await this.model.update(
+      { username, firstName, lastName, email },
+      {
+        where: {
+          id: this.id,
+        },
+        returning: true,
+      }
+    );
+
+    Object.assign(this, affectedRows[0].get());
+  }
+
+  async delete(
+    auth: Authenticator,
+    transaction?: Transaction
+  ): Promise<Result<undefined, Error>> {
+    try {
+      await this.model.destroy({
+        where: {
+          id: this.id,
+        },
+        transaction,
+      });
+
+      return new Ok(undefined);
+    } catch (err) {
+      return new Err(err as Error);
+    }
+  }
+
+  async unsafeDelete(
+    transaction?: Transaction
+  ): Promise<Result<undefined, Error>> {
+    try {
+      await this.model.destroy({
+        where: {
+          id: this.id,
+        },
+        transaction,
+      });
+
+      return new Ok(undefined);
+    } catch (err) {
+      return new Err(err as Error);
+    }
+  }
+
+  getAttribute(attribute: string): any {
+    switch (attribute) {
+      case "fullName":
+        return [this.firstName, this.lastName].filter(Boolean).join(" ");
+      default:
+        return "";
+    }
+  }
```

Comment:
IHMO over-engineered, a simple, would have done the job
 ```suggestion
  get fullName(){
      return [this.firstName, this.lastName].filter(Boolean).join(" ");
  }
```

Reply:
Misunderstood then 🙈